File: src/main/java/com/lcaohoanq/nocket/domain/chat/ChatRoomRepository.java
package com.lcaohoanq.nocket.domain.chat;

import com.lcaohoanq.nocket.domain.user.User;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ChatRoomRepository extends JpaRepository<ChatRoom, UUID> {
    
    Optional<ChatRoom> findByUser1AndUser2(User user1, User user2);

}

File: src/main/java/com/lcaohoanq/nocket/domain/chat/ChatRoomRepository.kt
package com.lcaohoanq.nocket.domain.chat

import com.lcaohoanq.nocket.domain.user.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ChatRoomRepository : JpaRepository<ChatRoom, UUID> {
    fun findByUser1AndUser2(user1: User, user2: User): Optional<ChatRoom>
}
